test/distributed/test_c10d.py

```diff
@@ -1673,14 +1672,20 @@ def allreduce(tensors, op):
         for i in range(self.num_gpus):
             self.assertEqual(torch.tensor([self.num_gpus]), tensors[i])
 
+        for op in (c10d.ReduceOp.BAND, c10d.ReduceOp.BOR, c10d.ReduceOp.BXOR):
+            with self.assertRaises(RuntimeError):
```

Comment:
Let's use `assertRaisesRegex` and pass in a regex string of the expected error message